src/main/java/net/imglib2/algorithm/fill/ComparableComparator.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * #%L
+ * ImgLib2: a general-purpose, multidimensional image processing library.
+ * %%
+ * Copyright (c) 2009 - 2016, Tobias Pietzsch, Stephan Preibisch, Stephan Saalfeld,
+ * John Bogovic, Albert Cardona, Barry DeZonia, Christian Dietz, Jan Funke,
+ * Aivar Grislis, Jonathan Hale, Philipp Hanslovsky, Grant Harris, Stefan Helfrich,
+ * Mark Hiner, Martin Horn, Steffen Jaensch, Lee Kamentsky, Larry Lindsey,
+ * Melissa Linkert, Mark Longair, Brian Northan, Nick Perry, Dimiter Prodanov,
+ * Curtis Rueden, Johannes Schindelin, Jean-Yves Tinevez and Michael Zinsmaier.
+ * All rights reserved.       
+ *                            
+ * Redistribution and use in source and binary forms, with or without modification,
+ * are permitted provided that the following conditions are met:
+ *                            
+ * 1. Redistributions of source code must retain the above copyright notice, this
+ *    list of conditions and the following disclaimer.
+ *                            
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *                            
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package net.imglib2.algorithm.fill;
+
+import net.imglib2.util.Pair;
+
+import java.util.Comparator;
+
+/**
+ * @author Philipp Hanslovsky &lt;hanslovskyp@janelia.hhmi.org&gt;
+ *
+ *
+ * Default implementation of {@link Comparable} for the use in {@link FloodFill} when source and target are both
+ * {@link net.imglib2.RandomAccessible<Comparable>}
+ *
+ */
+public class ComparableComparator< T extends Comparable< T >, U extends Comparable< U >  >
+        implements Comparator<Pair< T, U > > {
+
+    @Override
+    public int compare( final Pair< T, U > p1, final Pair< T, U > p2) {
+        int comp1 = p1.getA().compareTo( p2.getA() );
+        int comp2 = p1.getB().compareTo( p2.getB() );
+        return  comp1 == 0 ?
```

Comment:
Actually, my thoughts on XOR are only true, if comp1 and comp2 are binary, i.e. they always take the same value if non-zero. That makes this a little more copmlicated.